### PR TITLE
Validate Gregorian Date

### LINKED ISSUE #10057: `GregorianDate` constructor does not properly handle values
I found this issue while trying to convert a `GregorianDate` into a `JulianDate`. The [`GregorianDate` constructor](https://github.com/CesiumGS/cesium/blob/48c8abaca6257b568edc8c992ede5f2c487cdd38/Source/Core/GregorianDate.js) simply takes the parameters and puts them in `GregorianDate`'s respective fields, without actually validating them. For instance, the `hour` value should be in the range [0, 23], but nothing in the constructor actually enforces this.

Additionally, the [`GregorianDate` documentation](https://cesium.com/learn/cesiumjs/ref-doc/GregorianDate.html) says that all parameters can be optional. This seems strange -- what is the expected outcome of saying `var date = new GregorianDate()` ? It may make sense for some of these fields to be optional, like `second` and `millisecond`, but there is nothing in the constructor that provides a default value. This becomes problematic if I'm writing code along these lines:

```
var gregorianDate = new Cesium.GregorianDate(2022, 2, 1, 12, 0, 0);
var julianDate = new Cesium.JulianDate.fromGregorianDate(gregorianDate);
```

`JulianDate` expects the `GregorianDate` to have valid values. However, I didn't specify the `millisecond` value of the `GregorianDate`, so that is undefined. Thus, when `JulianDate` tries to perform this conversion, it results in a `NaN`.

### PR DESCRIPTION
**Pull request by jaredwebber**
Fixes #10057 

Reviving changes from #10075
Bulk of changes from above PR, I've gone ahead and resolved remaining PR comments 

# Description
Following are the use cases covered:
- Validate inclusive range for year (0001 to 9999)
- Validate inclusive range for month (1 to 12)
- Validate inclusive range for month (1 to 28/29/30/31 depending on month & year)
- Validate inclusive range for hour (0 to 23)
- Validate inclusive range for minute (0 to 59)
- Validate inclusive range for second (1 to 59/60 depending on isLeapSecond)
- Validate millisecond range from [0, 1000)

# Author checklist

- [X] I have submitted a Contributor License Agreement
- [X] I have added my name to `CONTRIBUTORS.md`
- [X] I have updated `CHANGES.md` with a short summary of my change
- [X] I have added or updated unit tests to ensure consistent code coverage
- [X] I have updated the inline documentation, and included code examples where relevant
- [X] I have performed a self-review of my code

### CODE REVIEW

**Comment by jaredwebber** [comment]
Pinging participants from original PR: @ggetz @j9liu

**Comment by ggetz** [comment]
Thanks @jaredwebber! This PR is looking good overall! 

Since we are releasing today, we will likely do a review later after the release is officially out. That means these changes will go out with the February 1 release.

**Review by ggetz — COMMENTED** [review]
Awesome, thanks for finishing up the work here @jaredwebber. Since the majority of the code has been previously reviewed, I don't have much else here in terms of comments or questions.

Since CI didn't run on this PR, I ran the build, validation, and unit testing locally. Everything is passing.

File: CHANGES.md [hunk 1]
```diff
@@ -13,6 +13,7 @@
 - Changes the default `RequestScheduler.maximumRequestsPerServer` from 6 to 18. This should improve performance on HTTP/2 servers and above [#11627](https://github.com/CesiumGS/cesium/issues/11627)
 - Corrected JSDoc and Typescript definitions that marked optional arguments as required in `ImageryProvider` constructor [#11625](https://github.com/CesiumGS/cesium/issues/11625)
 - The `Quaternion.computeAxis` function created an axis that was `(0,0,0)` for the unit quaternion, and an axis that was `(NaN,NaN,NaN)` for the quaternion `(0,0,0,-1)` (which describes a rotation about 360 degrees). Now, it returns the x-axis `(1,0,0)` in both of these cases.
+- Fixed a bug where `GregorianDate` constructor would not validate the input parameters for valid date. [#10075](https://github.com/CesiumGS/cesium/pull/10075)
```
[the remark this answers is not in the document]
Comment (jaredwebber) [reply CHANGES.md]: Moved changes to feb release ✅

**Comment by ggetz** [comment]
Awesome, thanks @jaredwebber!

**Pull request merged**